tests/007-datagrams.cpp

```diff
@@ -279,15 +298,317 @@ namespace oxen::quic::test
             REQUIRE(client->datagrams_enabled());
 
             REQUIRE(conn_interface->datagrams_enabled());
-            REQUIRE_FALSE(conn_interface->packet_splitting_enabled());
-            REQUIRE(conn_interface->packet_splitting_policy() == Splitting::NONE);
-            REQUIRE(conn_interface->get_max_datagram_size() < MAX_GREEDY_PMTUD_UDP_PAYLOAD);
+            REQUIRE(conn_interface->packet_splitting_enabled());
 
-            conn_interface->send_datagram(msg);
+            auto max_size = conn_interface->get_max_datagram_size();
+
+            std::string good_msg{}, oversize_msg{};
+            char v = 0;
+
+            while (good_msg.size() < max_size)
+                good_msg += v++;
+            v = 0;
+            while (oversize_msg.size() < max_size + 100)
+                oversize_msg += v++;
+
+            REQUIRE_NOTHROW(conn_interface->send_datagram(std::move(good_msg)));
+            REQUIRE_THROWS(conn_interface->send_datagram(std::move(oversize_msg)));
 
             REQUIRE(data_future.get());
+            REQUIRE(data_counter == 1);
+            test_net.close();
+        };
+    };
+
+    TEST_CASE(
+            "007 - Datagram support: Rotating Buffer, Clearing Buffer", "[007][datagrams][execute][split][rotating][clear]")
+    {
+        if (disable_rotating_buffer)
+            SKIP("Rotating buffer testing not enabled for this test iteration!");
+
+        SECTION("Simple oversized datagram transmission - Clear first row")
+        {
+            log::trace(log_cat, "Beginning the unit test from hell");
+            Network test_net{};
+
+            std::atomic<int> index{0};
+            std::atomic<int> data_counter{0};
+            size_t bufsize = 256, n = bufsize / 2 + 1;
```

Comment:
not a huge fan of multiple variable declarations on the same line.  also since `n` is used in several places and is not like a simple counter,  it could use a more descriptive name.